snuba/consumers/consumer.py

```diff
@@ -140,39 +159,50 @@ def close(self) -> None:
         )
         write_finish = time.time()
 
-        max_latency: Optional[float] = None
-        latency_sum = 0.0
-        max_end_to_end_latency: Optional[float] = None
-        end_to_end_latency_sum = 0.0
+        snuba_latency_recorder = LatencyRecorder()
+        end_to_end_latency_recorder = LatencyRecorder()
+        sentry_received_latency_recorder = LatencyRecorder()
+
         for message in self.__messages:
             assert isinstance(message.value, BrokerValue)
+
             latency = write_finish - message.value.timestamp.timestamp()
-            latency_sum += latency
-            if max_latency is None or latency > max_latency:
-                max_latency = latency
-            if message.payload.origin_timestamp is not None:
-                end_to_end_latency = (
-                    write_finish - message.payload.origin_timestamp.timestamp()
+            snuba_latency_recorder.record(latency)
+
+            origin_timestamp = message.payload.origin_timestamp
+            if origin_timestamp is not None:
+                end_to_end_latency = write_finish - origin_timestamp.timestamp()
+                end_to_end_latency_recorder.record(end_to_end_latency)
+
+            sentry_received_timestamp = message.payload.sentry_received_timestamp
+            if sentry_received_timestamp is not None:
+                sentry_received_latency = (
+                    write_finish - sentry_received_timestamp.timestamp()
                 )
-                end_to_end_latency_sum += end_to_end_latency
-                if (
-                    max_end_to_end_latency is None
-                    or end_to_end_latency > max_end_to_end_latency
-                ):
-                    max_end_to_end_latency = end_to_end_latency
-
-        if max_latency is not None:
-            self.__metrics.timing("max_latency_ms", max_latency * 1000)
+                sentry_received_latency_recorder.record(sentry_received_latency)
+
+        if snuba_latency_recorder.max is not None:
+            self.__metrics.timing("max_latency_ms", snuba_latency_recorder.max * 1000)
             self.__metrics.timing(
-                "latency_ms", (latency_sum / len(self.__messages)) * 1000
+                "latency_ms", (snuba_latency_recorder.sum / len(self.__messages)) * 1000
             )
-        if max_end_to_end_latency is not None:
+        if end_to_end_latency_recorder.max is not None:
```

Comment:
Rather than checking if `recorder.max is not None` maybe we could do `recorder.has_data()` (which can check if `max is not None`) which reveals less about the implementation to the caller